### PR TITLE
CRM-91: approach with dynamic parsing service name from resourceId

### PR DESCRIPTION
**Pull request by vagharshakus**
Approach with dynamic parsing service name from resourceId.

PR for previous solution closed, because considered to be sub-optimal: https://github.com/edekadigital/terraform-azure-modules/pull/3

Solution here should be quite robust, if consider structure convention of Azure resourceId.

Why?

- for cases, provider is a part of the resourceId, which should be the case for Azure Web and Function Apps,(MICROSOFT.WEB f.e. in case of Azure Web Apps and Functions App - s. also attached screenshot), service name should reliably be at last index of parsed Azure resourceId
- For all another cases, service is set to default "azure" value

In general, this should be enough for our purposes, because this kind of log forwarding (via event hub) is only applicable for Azure Web Apps and Function Apps (I speak mainly about monitorable Azure Compute Services). For VMs and Azure Containers another log shipping approaches are used, like Docker Agent / Fluent Bit Sidecar Container.

By the way, we use same approach in our AWS Lambda Log Forwarder, I took a look there.

azure_docu

### CODE REVIEW

**Comment by felixb** [comment]
I strongly advice not to do it like this. For example in our case, this would leak the stage into the tag making filtering harder than it should be.

**Comment by vagharshakus** [comment]
Why You think, stage will leak into service tag's value. At least, unit tests are not ondicating such behavioural. Maybe You can explain Your point on Monday.

**Comment by vagharshakus** [comment]
Ok. Now I understand, what You mean. You mean the "dev" / "test" infix in the service name. It's not "leaking", it's just the name of the Web App in Azure, as we called it. I don't consider to be this a problem. If ever, solution would be to rename the resource in Azure and make it uniquely "cat-app" in all subscrptions.

**Comment by vagharshakus** [comment]
Additional reason to do it like this for me is the obvious benefit, we bring into the solution in comparison with current Datadog Standard Solution (which is f.e. still used by Asteroids team - they compulsive will still have this problem). Maybe we can move then more teams / people to use this module.

File: terraform-azure-datadog-log-forwarder/azure-datadog-log-forwarder/src/log-forwarder.js [hunk 1]
```diff
@@ -303,6 +303,8 @@ class EventhubLogForwarder {
         if (resourceId[2] === "providers" && this.isSource(resourceId[3])) {
           // handle provider-only resource IDs
           metadata.source = this.formatSourceType(resourceId[3]);
+          // service name should be the last part of resourceID
```
Comment (selamanse) [inline terraform-azure-datadog-log-forwarder/azure-datadog-log-forwarder/src/log-forwarder.js:306]: can you provide the schematics for the resource ID here for future lookup?
(i.e. a link to the ms docs)

**Review by selamanse — APPROVED** [review]
change is ok and should do the trick for parsing the service name. A link to the resource.id schematic for further improvements would be nice.

**Comment by selamanse** [comment]
> I strongly advice not to do it like this. For example in our case, this would leak the stage into the tag making filtering harder than it should be.

I'd rather prefer removing the stage out of the app name and use resource tags within azure.

File: terraform-azure-datadog-log-forwarder/azure-datadog-log-forwarder/src/log-forwarder.js (hunk 1 above)
Comment (vagharshakus) [reply terraform-azure-datadog-log-forwarder/azure-datadog-log-forwarder/src/log-forwarder.js]: Good idea. There is no "central" MS docu about resourceId format for different Azure Resources. The best way to check it seems to be Azure CLI (**az resource list --name "myniceresource"**). Proper code inline comment added.

**Pull request merged**